apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/observer/TrimObserver.java

```diff
@@ -36,9 +42,39 @@
     private final Object lock = new Object();
     private final Queue<Integer> trimHeights = new PriorityQueue<>(); // will sort heights from lowest to highest automatically
     private ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+    private BlockchainConfig blockchainConfig;
+    private PropertiesHolder propertiesHolder;
+    private Random random;
+    private boolean isShardingOff;
+    private final int maxRollback;
+
+    @Inject
+    public TrimObserver(TrimService trimService, BlockchainConfig blockchainConfig,
+                        PropertiesHolder propertiesHolder, Random random) {
+        this.trimService = Objects.requireNonNull(trimService, "trimService is NULL");
+        this.blockchainConfig = Objects.requireNonNull(blockchainConfig, "blockchainConfig is NULL");
+        this.propertiesHolder = Objects.requireNonNull(propertiesHolder, "propertiesHolder is NULL");
+        this.trimFrequency = Constants.DEFAULT_TRIM_FREQUENCY;
+        this.isShardingOff = this.propertiesHolder.getBooleanProperty("apl.noshardcreate", false);
+        if (random == null) {
+            this.random = new Random();
+        } else {
+            this.random = random;
+        }
+        this.maxRollback = this.propertiesHolder.getIntProperty("apl.maxRollback", 720);
+    }
 
     @PostConstruct
     void init() {
+        if (this.blockchainConfig.getCurrentConfig() != null
+                && this.blockchainConfig.getCurrentConfig().getShardingFrequency() > 0 && this.trimFrequency > 0
```

Comment:
check `this.blockchainConfig.getCurrentConfig() != null` id neede in unit tests to prevent NPE, because initialization flow is different then in node.
added 'this.blockchainConfig.getCurrentConfig().isShardingEnabled()' as well

Reply:
You should mock currentConfig to prevent Npe, such  logic become more complicated and may be confusing in future

Reply:
CurrentConfig is already **mocked**, but components initialization flow is different in unit test then in node run-rime.